### PR TITLE
Extend REST API to include UserProjectACLResource.

### PR DESCRIPTION
**Pull request by carlohamalainen**
Adds functionality for adding AAF-type user accounts via the REST API.

Given the application-specific features (e.g. groups with the name "Project xyz") would this be better off as an app like the one James is doing for mydata?

### CODE REVIEW

**Comment by grischa** [comment]
Yeah, you can now include your own API hooks in an app.
See here: https://github.com/mytardis/mytardis/blob/develop/tardis/urls.py#L301
Basically, you need to name your resources 'blablaAppResource', in a file called api.py, then they will be available on the URL `/api/v1/appname_resourcename`. Alternatively, you can provide any kind of endpoint in your apps urls.py and views.py

File: tardis/tardis_portal/api.py [hunk 1]
```diff
@@ -1013,3 +1015,135 @@ def hydrate(self, bundle):
...
+                                     object_id=eps.experiment.id,
+                                     isOwner=False,
+                                     pluginId="django_group")
+                    oacl.save()
+
+
+        # Apply access for operators.
+        for eps in ExperimentParameterSet.objects.all():
+            operator_emails  = _get_value(eps, 'Operator')
+            instrument       = _get_value(eps, 'Instrument')
```
Comment (grischa) [inline tardis/tardis_portal/api.py]: Codacy Issue found: [Exactly one space required before assignment](https://www.codacy.com/app/mytardis/mytardis/file/630753152/issues/source?bid=2125231&fileBranchId=2304289#l1110)

File: tardis/tardis_portal/api.py [hunk 2]
```diff
@@ -1013,3 +1015,135 @@ def hydrate(self, bundle):
...
+                                     entityId=str(g.id),
+                                     object_id=eps.experiment.id,
+                                     isOwner=False,
+                                     pluginId="django_group")
+                    oacl.save()
+
+
+        # Apply access for operators.
+        for eps in ExperimentParameterSet.objects.all():
+            operator_emails  = _get_value(eps, 'Operator')
```
Comment (grischa) [inline tardis/tardis_portal/api.py]: Codacy Issue found: [Exactly one space required before assignment](https://www.codacy.com/app/mytardis/mytardis/file/630753152/issues/source?bid=2125231&fileBranchId=2304289#l1109)

File: tardis/tardis_portal/api.py [hunk 3]
```diff
@@ -1013,3 +1015,135 @@ def hydrate(self, bundle):
...
+
+            # Add this user to the group:
+            u.groups.add(g)
+            u.save()
+
+        # Apply ACLs for experiments.
+        for eps in ExperimentParameterSet.objects.all():
+            for p in eps.parameters:
+                str(p) # force calculation of _name_cache
+                if p._name_cache.name != 'Project': continue
```
Comment (grischa) [inline tardis/tardis_portal/api.py]: Codacy Issue found: [More than one statement on a single line](https://www.codacy.com/app/mytardis/mytardis/file/630753152/issues/source?bid=2125231&fileBranchId=2304289#l1085)

**Comment by carlohamalainen** [comment]
This code is now in my imagetrove plugin, so closing the PR.

**Pull request closed without merging**